### PR TITLE
chore(msk-alpha): update KafkaVersion

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

Could not find any in the backlog

### Reason for this change

Update the CDK listed Kafka versions to match the current availability, as well as add missing deprecated versions

### Description of changes

* Added latest version
	* `3.6.0` also supports tiered storage, see [docs](https://docs.aws.amazon.com/msk/latest/developerguide/msk-tiered-storage.html#msk-tiered-storage-requirements). I replaced the prefix check with a list of supported versions, as I'm not sure if say every reason after 3.6.0 will support it, and the `.tiered` prefix isn't consistently applied anymore
* Added two unlisted, deprecated versions, as they are still returned by the SDK. The goal is to remove any differences between the SDK and the CDK to ease future automation

### Description of how you validated changes

I compared the current CDK versions to live SDK data, using the `kafka:ListKafkaVersions` API results.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by msambol — APPROVED** [review]
LGTM. From the docs:

`The Amazon MSK cluster with tiered storage enabled must use version 3.6.0 or 2.8.2.tiered.`

File: packages/@aws-cdk/aws-msk-alpha/lib/cluster-version.ts [hunk 1]
```diff
@@ -119,6 +142,16 @@ export class KafkaVersion {
     return new KafkaVersion(version);
   }
 
+  /**
+   * List of Kafka versions that support tiered storage
+   *
+   * @see https://docs.aws.amazon.com/msk/latest/developerguide/msk-tiered-storage.html#msk-tiered-storage-requirements
+   */
+  private static readonly TIERED_STORAGE_COMPATIBLE_VERSIONS = [
```
Comment (GavinZZ) [inline packages/@aws-cdk/aws-msk-alpha/lib/cluster-version.ts:150]: Can we also update the [README](packages/@aws-cdk/aws-msk-alpha/lib/cluster-version.ts) here to mention that 3.6.0 also supports tiered storage.

**Review by GavinZZ — CHANGES_REQUESTED** [review]
Overall LGTM just one README update comment

File: packages/@aws-cdk/aws-msk-alpha/lib/cluster-version.ts (hunk 1 above)
Comment (nmussy) [reply packages/@aws-cdk/aws-msk-alpha/lib/cluster-version.ts:150]: Done, thanks

**Review by GavinZZ — DISMISSED** [review]
Thanks for contributing!

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**